feat: [PKD-009][PKD-010] [UI][Implementation] Type Filter Modal

Enrickzz opened the pull request:
-Pokemon list remove pagination (removed unnecessary variables/functions)
-Added getFiltered for filtered list
-Added widgets for filter_bottom_sheet
-Added floatingActionButton for filter modal
-Added GetFilterList action

https://user-images.githubusercontent.com/61441278/182323226-6a384579-c9e1-43b0-9fae-27e9ca35583e.mp4

Enrickzz committed:
feat: [PKD-009][PKD-010] [UI][Implementation] Type Filter Modal

-Pokemon list remove pagination (removed unnecessary variables/functions)
-Added getFiltered for filtered list
-Added widgets for filter_bottom_sheet
-Added floatingActionButton for filter modal
-Added GetFilterList action

dbenedictos commented:
please compare your UI to the template 😀

![Screenshot 2022-08-02 at 5 02 40 PM](https://user-images.githubusercontent.com/47196695/182336687-405914e4-ee64-4c3b-b996-1a5ced54756d.png)
![Screenshot 2022-08-02 at 5 03 33 PM](https://user-images.githubusercontent.com/47196695/182336795-74bb367c-778b-4873-8cc0-04f2e061d3cf.png)

dbenedictos commented on lib/feature/home_page/home_page_vm.dart:
```diff
@@ -8,10 +8,7 @@ import 'package:pokedex_async_redux/state/app_state.dart';
-      getPokemon: _getPokemon,
-      homePageState: _getPageState(),
-    );
+    return HomePageVm(homePageState: _getPageState());
```
=>

dbenedictos commented on lib/feature/home_page/widgets/filter_bottom_sheet_vm.dart:
```diff
@@ -0,0 +1,19 @@
+
+class FilterBottomSheetVmFactory extends VmFactory<AppState, FilterBottomSheetConnector> {
+  @override
+  Vm fromStore() {
```
=>

dbenedictos commented on lib/state/actions/home_page_actions.dart:
```diff
@@ -41,3 +36,18 @@ class GetEvolution extends LoadingAction {
 }
+
+class GetFilterList extends LoadingAction {
+  static const key = 'get_filtered_list_key';
```
not being used properly. There's no loading indication when this is called.

https://user-images.githubusercontent.com/47196695/182338288-b77c101c-8923-451c-b376-60c7cc6274e9.mov

dbenedictos commented on lib/feature/home_page/widgets/filter_bottom_sheet_vm.dart:
```diff
@@ -0,0 +1,19 @@
+    return FilterBottomSheetVm(onFilterTap: _onFilterTap);
+  }
+
+  void _onFilterTap(String filterType) => dispatch(GetFilterList(filterKey: filterType));
```
do your action segmentation here, instead of doing it inside the `GetFilterList`

Enrickzz committed:
PR Feedback

-UnionPageState proper Implementation
-Repositioned widgets in stack to put pokemon image behind pokemon name

dbenedictos commented on lib/feature/home_page/widgets/filter_bottom_sheet_vm.dart:
```diff
@@ -0,0 +1,29 @@
+class FilterBottomSheetVm extends Vm {
+  FilterBottomSheetVm({required this.onFilterTapState}) : super(equals: []);
+
+  final UnionPageState<Function(String filterType)> onFilterTapState;
```
ahhh no. UnionPageState is used when you're waiting for some data.

dbenedictos replied:
just pass the function as normal

dbenedictos commented on lib/state/actions/home_page_actions.dart:
```diff
@@ -41,3 +36,14 @@ class GetEvolution extends LoadingAction {
 }
+
+class GetFilterList extends LoadingAction {
+  static const key = 'get_filtered_list_key';
```
you're still not waiting for this

raelsj committed:
Create lints_workflow.yml

raelsj committed:
Update lints_workflow.yml

Enrickzz committed:
PR Feedback

-Added new page for filtered pokemons
-Updated dependencies
-Updated bottomsheet filter function onTap

Enrickzz committed:
Merge branch 'feature/dev_009' of github.com:Enrickzz/pokedex-async-redux into feature/dev_009

Enrickzz committed:
PR Feedback

-removed '@JsonKey' annotation

Enrickzz committed:
PR Feedback

-accidentally removed jsonkey

dbenedictos commented on lib/state/app_state.dart:
```diff
@@ -10,11 +10,12 @@ part 'app_state.g.dart';
+    EvolutionChain? evolution,
+    @Default(<Pokemon>[]) List<Pokemon> filteredPokemons,
+    @Default('') String filterKey,
+    @Default(Wait.empty) Wait wait,
```
retain the JsonKey here. since you need to ignore the wait

dbenedictos replied:
it's causing an error when generating the freezed and g files

dbenedictos commented on lib/feature/home_page/widgets/pokemon_filtered_page.dart:
```diff
@@ -0,0 +1,59 @@
+import 'package:pokedex_async_redux/api/models/pokemon_model.dart';
```
Filtered page is another feature, so please move it outside the home_page directory

dbenedictos commented on lib/feature/home_page/widgets/pokemon_filtered_page.dart:
```diff
@@ -0,0 +1,59 @@
+                      ))
+                  .toList(),
+            ),
+          ))
```
trailing comma

Enrickzz committed:
PR Feedback

-moved pokemon_filtered_page
-trailing commas

dbenedictos commented on lib/feature/home_page/widgets/filter_bottom_sheet.dart:
```diff
@@ -0,0 +1,64 @@
+            },
+          ),
+        )
+      ]),
```
trailing comma

dbenedictos commented on lib/feature/pokemon_filtered_page/pokemon_filtered_connector.dart:
```diff
@@ -0,0 +1,29 @@
+            loading: () => PokemonListLoading(),
+            error: (err) => Center(child: Text(err!)),
+          );
+        });
```
trailing comma

dbenedictos commented on lib/feature/home_page/widgets/filter_bottom_sheet.dart:
```diff
@@ -0,0 +1,64 @@
+            itemCount: colorFilters.length,
+            itemBuilder: (BuildContext context, int index) {
+              return GestureDetector(
+                onTap: () => Navigator.push(
```
pop the modal before navigating

dbenedictos commented on lib/state/actions/home_page_actions.dart:
```diff
@@ -41,3 +36,14 @@ class GetEvolution extends LoadingAction {
+  @override
+  Future<AppState> reduce() async {
+    final filteredRes = await PokemonHandler.getFiltered(filterKey);
+    return state.copyWith(filteredPokemons: filteredRes!, filterKey: filterKey);
```
trailing comma

Enrickzz committed:
PR Feedback

-Pop modal before navigate
-trailing commas
